localstack/services/awslambda/provider.py

```diff
@@ -218,6 +219,44 @@ def __init__(self) -> None:
         self.layer_fetcher = None
         lambda_hooks.inject_layer_fetcher.run(self)
 
+    def accept_state_visitor(self, visitor: StateVisitor):
+        visitor.visit(lambda_stores)
+
+    def on_before_state_load(self):
+        self.lambda_service.stop()
+
+    def on_after_state_load(self):
+        self.lambda_service = LambdaService()
+
+        # TODO: provisioned concurrency
+        for account_id, account_bundle in lambda_stores.items():
+            for region_name, state in account_bundle.items():
+                for fn in state.functions.values():
+                    for fn_version in fn.versions.values():
+                        # restore the "Pending" state for every function version and start it
+                        try:
+                            new_state = VersionState(
+                                state=State.Pending,
+                                code=StateReasonCode.Creating,
+                                reason="The function is being created.",
+                            )
+                            new_config = dataclasses.replace(fn_version.config, state=new_state)
```

Comment:
minor side-effect: might create a new `revision_id` because the state is included in the model